src/IROperator.cpp

```diff
@@ -1421,6 +1421,135 @@ Expr fast_cos(const Expr &x_full) {
     return fast_sin_cos(x_full, false);
 }
 
+// A vectorizable atan and atan2 implementation.
+// Based on the ideas presented in https://mazzo.li/posts/vectorized-atan2.html.
+Expr fast_atan_approximation(const Expr &x_full, ApproximationPrecision precision, bool between_m1_and_p1) {
+    const float pi_over_two = 1.57079632679489661923f;
+    Expr x;
+    // if x > 1 -> atan(x) = Pi/2 - atan(1/x)
+    Expr x_gt_1 = abs(x_full) > 1.0f;
+    if (between_m1_and_p1) {
+        x = x_full;
+    } else {
+        x = select(x_gt_1, 1.0f / x_full, x_full);
+    }
+
+    // Coefficients obtained using src/polynomial_optimizer.py
+    // Note that the maximal errors are computed with numpy with double precision.
+    // The real errors are a bit larger with single-precision floats (see correctness/fast_arctan.cpp).
+
+    // The table is huge, so let's put clang-format off and handle the layout manually:
+    // clang-format off
+    std::vector<float> c;
+    switch (precision) {
```

Comment:
Nit: since the enum here is monotonic, this might be more efficient as a static const LUT rather than a switch-with-assignment?

Reply:
I used switch to make sure compiler will error if I don't implement all cases.

Reply:
Performance isn't an issue either. This is run once when you call the function.

Reply:
Switch actually isn't monotonic: many have shared coefficients with fall through.